src/api/app/controllers/webui/watched_items_controller.rb

```diff
@@ -0,0 +1,40 @@
+class Webui::WatchedItemsController < Webui::WebuiController
+  before_action :check_user_belongs_feature_flag
+  before_action :set_item
+
+  FLASH_PER_WATCHABLE_TYPE = {
+    Package => 'package',
+    Project => 'project',
+    BsRequest => 'request'
+  }.freeze
+
+  def toggle
+    watched_item = User.session!.watched_items.find_by(watchable: @item)
```

Comment:
If you require people to be logged in, use `before_action :require_login`

Reply:
Using Pundit would be better. If you decide to go with `before_action :require_login`, please add this controller to the list of controllers to be refactored in #10083.

Reply:
Thanks for the warning. Added to the list in #10083.